File: sampleApp/src/main/java/com/nytimes/android/external/playbillingtester/sample/PrefsManager.java
package com.nytimes.android.external.playbillingtester.sample;

import android.content.SharedPreferences;

public class PrefsManager {

    private static final String KEY_IS_TEST_PROVIDER = "isTestProvider";
    private final SharedPreferences sharedPreferences;

    public PrefsManager(SharedPreferences sharedPreferences) {
        this.sharedPreferences = sharedPreferences;
    }

    public boolean isUsingTestGoogleServiceProvider() {
        return sharedPreferences.getBoolean(KEY_IS_TEST_PROVIDER, false);
    }

    public void setUsingGoogleServiceProvider(boolean isTest) {
        SharedPreferences.Editor edit = sharedPreferences.edit();
        edit.putBoolean(KEY_IS_TEST_PROVIDER, isTest);
        edit.commit();
    }

}

File: sampleApp/src/main/java/com/nytimes/android/external/register/sample/PrefsManager.kt
package com.nytimes.android.external.register.sample

import android.content.SharedPreferences

class PrefsManager(private val sharedPreferences: SharedPreferences) {

    val isUsingTestGoogleServiceProvider: Boolean
        get() = sharedPreferences.getBoolean(KEY_IS_TEST_PROVIDER, false)

    fun setUsingGoogleServiceProvider(isTest: Boolean) {
        val edit = sharedPreferences.edit()
        edit.putBoolean(KEY_IS_TEST_PROVIDER, isTest)
        edit.commit()
    }

    companion object {

        private val KEY_IS_TEST_PROVIDER = "isTestProvider"
    }

}
